x-pack/plugins/security_solution/server/endpoint/endpoint_app_context_services.ts

```diff
@@ -104,92 +105,128 @@ export class EndpointAppContextService {
       throw new EndpointAppContentServicesNotSetUpError();
     }
 
+    const savedObjectsFactory = new SavedObjectsClientFactory(
+      dependencies.savedObjectsServiceStart,
+      this.setupDependencies.httpServiceSetup
+    );
+
     this.startDependencies = dependencies;
     this.security = dependencies.security;
-    this.fleetServicesFactory = dependencies.endpointFleetServicesFactory;
+    this.savedObjectsFactoryService = savedObjectsFactory;
+    this.fleetServicesFactory = new EndpointFleetServicesFactory(
+      dependencies.fleetStartServices,
+      savedObjectsFactory
+    );
+
+    this.registerFleetExtensions();
+    this.registerListsExtensions();
+  }
+
+  public stop() {
+    this.startDependencies = null;
+    this.savedObjectsFactoryService = null;
+  }
+
+  private registerListsExtensions() {
+    if (this.startDependencies?.registerListsServerExtension) {
+      registerListsPluginEndpointExtensionPoints(
+        this.startDependencies?.registerListsServerExtension,
+        this
+      );
+    }
+  }
+
+  private registerFleetExtensions() {
+    if (!this.setupDependencies) {
+      throw new EndpointAppContentServicesNotSetUpError();
+    }
+    if (!this.startDependencies) {
+      throw new EndpointAppContentServicesNotStartedError();
+    }
+
+    const {
+      fleetStartServices: { registerExternalCallback: registerFleetCallback },
+      manifestManager,
+      alerting,
+      licenseService,
+      exceptionListsClient,
+      featureUsageService,
+      esClient,
+      productFeaturesService,
+    } = this.startDependencies;
+    const endpointMetadataService = this.getEndpointMetadataService();
+    const soClient = this.savedObjects.createInternalScopedSoClient(undefined, false);
```

Comment:
what do you think about changing  `createInternalScopedSoClient` to accept an object, instead of undefined passing undefined to get a default value?

Reply:
yeah, I can do that. Will push through a change soon